plugins/distance_table.hpp

```diff
@@ -97,21 +107,49 @@ template <class DataFacadeT> class DistanceTablePlugin final : public BasePlugin
                 ObjectEncoder::DecodeFromBase64(route_parameters.hints[i], current_phantom_node);
                 if (current_phantom_node.is_valid(facade->GetNumberOfNodes()))
                 {
-                    phantom_node_vector[i].emplace_back(std::move(current_phantom_node));
+                    phantom_node_target_vector[i].emplace_back(std::move(current_phantom_node));
                     continue;
                 }
             }
             const int bearing = input_bearings.size() > 0 ? input_bearings[i].first : 0;
             const int range = input_bearings.size() > 0 ? (input_bearings[i].second?*input_bearings[i].second:10) : 180;
-            facade->IncrementalFindPhantomNodeForCoordinate(route_parameters.coordinates[i],
-                                                            phantom_node_vector[i], 1, bearing, range);
+            facade->IncrementalFindPhantomNodeForCoordinate(useSameTgtSrc ? route_parameters.coordinates[i] : route_parameters.destinations[i],
+                                                            phantom_node_target_vector[i], 1, bearing, range);
 
-            BOOST_ASSERT(phantom_node_vector[i].front().is_valid(facade->GetNumberOfNodes()));
+            BOOST_ASSERT(phantom_node_target_vector[i].front().is_valid(facade->GetNumberOfNodes()));
+        }
+        unsigned shift_coordinates = (useSameTgtSrc) ? 0 : route_parameters.destinations.size();
+        max_locations = 0;
+        if (!useSameTgtSrc)
+        {
+            max_locations = std::min(static_cast<unsigned>(max_locations_distance_table),
+                                     static_cast<unsigned>(route_parameters.sources.size()));
+        }
+        PhantomNodeArray phantom_node_source_vector(max_locations);
+        for (const auto i : osrm::irange(0u, max_locations))
+        {
+            if (checksum_OK && i < route_parameters.hints.size() - shift_coordinates &&
+                !route_parameters.hints[i + shift_coordinates].empty())
+            {
+                PhantomNode current_phantom_node;
+                ObjectEncoder::DecodeFromBase64(route_parameters.hints[i + shift_coordinates], current_phantom_node);
```

Comment:
We sadly need to drop `hint` support here. `5.0.0` will restructure the API a little, I think we can live with this for one release cycle.